src/clients/Client.cc

```diff
@@ -522,6 +526,7 @@ Client::blockCaching()
         // That method lacks HttpRequest and may be called too many times.
         ACLFilledChecklist ch(acl, originalRequest().getRaw());
         ch.reply = const_cast<HttpReply*>(entry->getReply()); // ACLFilledChecklist API bug
+        ch.al = fwd->al;
```

Comment:
Please do not separate the `ch.reply` assignment above from its HTTPMSGLOCK() below.